src/main/java/org/apache/sysds/runtime/util/DataConverter.java

```diff
@@ -307,6 +307,34 @@ public static MatrixBlock readMatrixFromHDFS(ReadProperties prop)
 		return ret;
 	}
 
+	public static List<Integer> convertVectorToIndicesList(MatrixBlock mb)
+	{
+		int rows = mb.getNumRows();
+		int cols = mb.getNumColumns();
+		ArrayList<Integer> indices = new ArrayList<>();
+
+		if( mb.getNonZeros() > 0 )
```

Comment:
first if check for empty. (return null)
second check for sparse (return already allocated the sparse allocated indexes )
third (default) iterate dense. (allocate based on nnz)